stores/planship.ts

```diff
@@ -0,0 +1,258 @@
+    useRuntimeConfig().public.webSocketUrl,
+  )
+}
+
+function createBrowserApiClient() {
+  return new Planship('clicker-demo', getApiUrl(), getAccessToken, useRuntimeConfig().public.webSocketUrl)
+}
+
+// This class should come autogenerated by Planship (as a snippet to copy/paste?)
+export class Entitlements {
+
+  private entitlementsDict: {} = {
+    'subscription-button-clicks': 0,
+    'button-clicks-per-minute': 0,
+    'max-projects': 0,
+    'premium-button': false,
+    'analytics-panel': false,
+    'project-types': [ 'Single' ]
+  }
+
+
+  constructor(entitlementsDict?: {}) {
+    if (entitlementsDict)
+      this.entitlementsDict = entitlementsDict;
+  }
+
+  get subscriptionButtonClicks(): number {
+    return this.entitlementsDict['subscription-button-clicks']
+  }
+
+  get buttonClicksPerMinute(): number {
+    return this.entitlementsDict['button-clicks-per-minute']
+  }
+
+  get maxProjects(): number {
+    return this.entitlementsDict['max-projects']
+  }
+
+  get premiumButton(): boolean {
+    return this.entitlementsDict['premium-button']
+  }
+
+  get analyticsPanel(): boolean {
+    return this.entitlementsDict['analytics-panel']
+  }
+
+  get projectTypes(): [] {
+    return this.entitlementsDict['project-types']
+  }
+}
+
+export const usePlanshipStore = defineStore('planship', () => {
+
+    const subscriptions = ref([])
+    const entitlementsDict = ref({})
+    const plans = ref([])
+    const clickAnalytics = ref({})
+    const currentUser = ref({})
+
+
+    const userStore = useUserStore()
+    const projectsStore = useProjectsStore()
+    const { ssrContext } = useNuxtApp();
+    const apiClient = ssrContext ? createServerApiClient() : createBrowserApiClient();
+
+    const defaultSubscription = computed(() => subscriptions.value.length > 0 ? subscriptions.value[0] : undefined)
+
+    const entitlements = computed(() => new Entitlements(entitlementsDict.value))
+
+    const currentPlanSlug = computed(() => {
+      return defaultSubscription.value ? defaultSubscription.value.plan.slug : ''
+    })
+
+    const currentPlanName = computed(() => defaultSubscription.value ? defaultSubscription.value.plan.name : '')
+
+    const canCreateProject = computed(() => {
+      return entitlements.value.maxProjects > projectsStore.projects.length
+    })
+
+    const canGenerateButtonClick = computed(() => entitlements.value.subscriptionButtonClicks > 0 && entitlements.value.buttonClicksPerMinute > 0)
+
+    function updateEntitlementsCb(entitlements) {
+      entitlementsDict.value = entitlements
+    }
+
+    async function createCustomerAndSubscription(userid, planSlug) {
+      try {
+        // Create a Planship customer
+        const user = await apiClient.createCustomer({alternativeId: userid})
+        // Create a default plan subscription for the new customer
+        await apiClient.createSubscription(user.id, planSlug)
+        return user
+      } catch(error) {
+        console.dir(error.response)
```

Comment:
Add a comment that this is where you'd want to handle a failure scenario. Could set up a simple `ErrorHandler` lib that takes the error and prints it to get the idea across that this is how you handle errors.